pkg/sdk/identifier_parsers.go

```diff
@@ -140,3 +145,50 @@ func ParseExternalObjectIdentifier(identifier string) (ExternalObjectIdentifier,
 		},
 	)
 }
+
+// ParseFunctionArgumentsFromString parses function argument from arguments string with optional argument names.
+// Varying types are not supported (e.g. VARCHAR(200)), because Snowflake outputs them in shortened version
+// (VARCHAR in this case). The only exception is newly added type VECTOR which has the following structure
+// VECTOR(<type>, n) where <type> right now can be either INT or FLOAT and n is the number of elements in the VECTOR.
+// Snowflake returns vectors with their exact type and this function supports it.
+func ParseFunctionArgumentsFromString(arguments string) ([]DataType, error) {
+	dataTypes := make([]DataType, 0)
+
+	if len(arguments) > 0 && arguments[0] == '(' && arguments[len(arguments)-1] == ')' {
+		arguments = arguments[1 : len(arguments)-1]
+	}
+	stringBuffer := bytes.NewBufferString(arguments)
+
+	for stringBuffer.Len() > 0 {
+		// We use another buffer to peek into next data type (needed for vector parsing)
+		peekDataType, _ := bytes.NewBufferString(stringBuffer.String()).ReadString(',')
+		peekDataType = strings.TrimSpace(peekDataType)
+
+		switch {
+		// For now, only vectors need special parsing behavior
+		case strings.HasPrefix(peekDataType, "VECTOR"):
+			vectorDataType, _ := stringBuffer.ReadString(')')
```

Comment:
Why is this error not handled?

Reply:
As documentation says, the only error we can get here is EOF which also happens when the whole buffer is read, so let's say we have `(abc)` and you want to call `stringBuffer.ReadString(')')` it will return EOF because the function reached the end of buffer. That's why I decided to skip them, because there's no other error than EOF and it happens in cases where I don't expect it to. I'll leave additional comments for context.

Reply:
Actually, I was wrong the error is only returned if the returned string is not ending with the expected delimiter. I'll add err checks and test cases